### DISCUSSION TITLE
Help with understanding the ssGSEA computations

### DISCUSSION THREAD

**Discussion by erolkavvas** [zqfang/GSEApy#173] [Q&A] [answered]
Hi,

I'm basically trying to replicate the results of ssGSEA with my own code and have not been able to. I was wondering if you guys could help me and point out what I'm missing. My pipeline is as follows, 

1. I start with data normalization with the "rank" approach
2. Sort the data by the normalized ranks (_load_ranking() in GSEAbase())
3. Then compute the enrichment score (basically copying enrichment_score() with ssGSEA parameters)

My results are the same as when I use the gseapy functions with this pipeline. However, the result is different from when I run the single ssgsea() function, which means I must be missing a step.

I haven't been able to follow the code in this repo all the way through because I'm unable to find the gseapy.gse file, which the ssgsea() function calls. It's a ".so" file on my computer and won't open.

Thank you,
Erol

**Comment by zqfang** [comment, marked as answer]
The code you'er looking for is here(line 668 - 687)

https://github.com/zqfang/GSEApy/blob/cbcb701642f7e79150c19ccda6e666f2d882102b/src/stats.rs#L668


in short, the Rust implementation add z_score normalization step as ssGSEA did. see the line comments. 

The  `enrichment_score()` did not do the z-score normalization. That's the only difference between the two

**Reply by erolkavvas** [reply]
Thank you!
